client/client-multi/client-c/src/platform-impl/kaa_tcp_channel.c

```diff
@@ -0,0 +1,1454 @@
+/*
+ * Copyright 2014-2015 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+#include <time.h>
+
+#include "../platform/stdio.h"
+
+#include "../kaa_common.h"
+#include "../utilities/kaa_mem.h"
+#include "../utilities/kaa_buffer.h"
+#include "../utilities/kaa_log.h"
+#include "../kaa_protocols/kaa_tcp/kaatcp.h"
+#include "../platform/ext_system_logger.h"
+#include "kaa_tcp_channel.h"
+
+#define KAA_TCP_CHANNEL_IN_BUFFER_SIZE     1024
+#define KAA_TCP_CHANNEL_OUT_BUFFER_SIZE    2024
+
+#define KAA_TCP_CHANNEL_TRANSPORT_PROTOCOL_ID         0x56c8ff92
+#define KAA_TCP_CHANNEL_TRANSPORT_PROTOCOL_VERSION    1
+
+#define KAA_TCP_CHANNEL_NEXT_PROTOCOL_BINARY_ID       0x3553c66f
```

Comment:
This does not belong here. It must be defined in the binary platform level protocol.